crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -228,16 +313,61 @@ fn split_recipients_withhelds_for_user(
         user_devices.into_values().partition_map(|d| {
             if d.is_blacklisted() {
```

Comment:
This if/else sequence is getting harder and harder to read with all the possible settings/outcome.
We might want to get rid of the partition/Either system.
Also the set of settings can create invalid configuration, like `only_allow_trusted_devices`  + `error_on_unsigned_device_of_verified_users` doesn't really make sense.
Will get worse with more options